test/symforce_function_codegen_test_data/symengine/symforce_rust_codegen_test/src/backend_test_function_float32.rs

```diff
@@ -0,0 +1,305 @@
+
+// -----------------------------------------------------------------------------
+// This file was autogenerated by symforce from template:
+//     function/FUNCTION.rs.jinja
+// Do NOT modify by hand.
+// -----------------------------------------------------------------------------
+
+mod sym {
+
+    use nalgebra as na;
+
+    ///
+    /// Given input symbols `x` and `y`, return a list of expressions which provide good test coverage
+    /// over symbolic functions supported by symforce.
+    ///
+    /// The intention is that generating this function for a given backend should provide good test
+    /// coverage indicating that the printer for that backend is implemented correctly.
+    ///
+    /// This does not attempt to test the rest of the backend (any geo, cam, matrix, or DataBuffer use),
+    /// just the printer itself.
+
+    pub fn backend_test_function_float32(
+        x: f32,
+        y: f32,
+        res0: Option<&mut f32>,
```

Comment:
I don't love Option<&mut Type> as the method for returning a variable but I wanted it to look as much like the cpp version as possible. If we can remove the restriction I can try to come up with a more rust-y way of doing this.

Reply:
Yeah, I think we should do things idiomatically for rust, I wouldn't be worried about matching cpp.  For cpp we support both return values and output arguments, and in particular output arguments can be optional, and some compute is saved if they aren't requested.  It might be nice to support that, but also I think it'd be fine to always compute all outputs for now.

If we do want to support optional outputs, naively `Option<&mut T>` seems reasonable to me, although I don't quite know enough rust to know if there's a more idiomatic way

Reply:
Yeah I think that's ok, especially since the Rust backend is marked as experimental for now, although we should try and avoid that if we can